src/integrations/cleanup-integration.php

```diff
@@ -520,4 +530,34 @@ protected function cleanup_orphaned_from_table( $table, $column, $limit ) {
 		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared, WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Reason: Already prepared.
 		return $wpdb->query( "DELETE FROM $table WHERE {$column} IN( " . \implode( ',', $orphans ) . ' )' );
 	}
+
+	/**
+	 * Deletes rows from the indexable table where the source is no longer there.
+	 *
+	 * @param string $source_table The source table which we need to check the indexables against.
+	 * @param string $source_identifier The identifier which the indexables are matched to.
+	 * @param string $object_type The indexable object type.
+	 * @param int    $limit The limit we'll apply to the delete query.
+	 *
+	 * @return int|bool The number of rows that was deleted or false if the query failed.
+	 */
+	protected function clean_indexables_for_object_type_and_source_table( $source_table, $source_identifier, $object_type, $limit ) {
+		global $wpdb;
+
+		$indexable_table = Model::get_table_name( 'Indexable' );
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: There is no unescaped user input.
+		$delete_query = $wpdb->prepare(
+			"DELETE FROM $indexable_table
+				WHERE object_type = '$object_type'
+				AND object_id NOT IN (
+					SELECT $source_identifier FROM $source_table
+				) LIMIT %d",
```

Comment:
Was it a conscious decision to have the orphaned records be deleted in this way and not by first gathering the indexable records with a LEFT JOIN and then deleting them in a separate query (basically, what we do with `cleanup_orphaned_from_table`)? 

If not, I think we should investigate more (maybe with EXPLAINs) on which option is the best performance-wise, here.

Reply:
For the above, maybe compare the current solution, with having 2 separate queries.

One for gathering the orphaned data, which could look something like this (quickly tested, needs more careful look):
```
		$query = $wpdb->prepare(
			"
			SELECT indexable_table.object_id
			FROM {$indexable_table} indexable_table
			LEFT JOIN {$source_table} AS source_table
			ON indexable_table.object_id = source_table.{$source_identifier}
			WHERE source_table.{$source_identifier} IS NULL
			AND indexable_table.object_id IS NOT NULL
			AND indexable_table.object_type = '{$object_type}'
			LIMIT %d",
			$limit
		);
```
and one for deleting them, like:
```
return $wpdb->query( "DELETE FROM $indexable_table WHERE object_id IN( " . \implode( ',', $orphans ) . ' )' );
```

Reply:
This makes sense will pivot to this approach to keep everything consistent. This one is more save so I think we won't need to do explains